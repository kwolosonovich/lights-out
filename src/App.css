@import url("https://fonts.googleapis.com/css2?family=Orbitron:wght@700&display=swap");

html {
  min-height: 100%;
  min-width: 100%;
  background-image: url(./4.jpg);
  background-position: center;
  background-size: cover;
}

.flex-container {
  margin: 50px 30px 0px 30px;
  display: flex;
  height: 700px;
  flex-wrap: wrap;
  align-content: space-between;
}

.App {
  display: inline-flex;
  text-align: center;
  font-family: "Orbitron", sans-serif;
  background-image: url(./1.jpg);
  color: rgba(255, 255, 255, 0.726);
  font-size: 60px;
  width: 50%;
  min-width: 300px;
  border-radius: 15px;
  box-shadow: 2px 2px 4px 4px rgba(0, 0, 0, 0.212);
}

.col2 {
  display: inline-flex;
  width: 50%;
}
