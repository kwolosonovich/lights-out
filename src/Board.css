.game {
  display: block;
  width: 100%;
  text-align: center;
  font-size: 2.2rem;
  align-items: center;
  margin-bottom: 30px;
}

table {
  position: relative;
  align-items: center;
  margin: auto;
  padding: .5rem;
} 

tbody {
  position: relative;
  width: 100%;
  margin-left: 50%;
}

