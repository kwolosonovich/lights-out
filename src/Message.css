h2 {
    width: 100%;
    margin-top: 200px;
}

button {
    display: block;
    height: 15%;
    width: 100%;
}

.restart {
    display: block;
    background-color: rgba(255, 255, 255, 0);
    border: none;
}
