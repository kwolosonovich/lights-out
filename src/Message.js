import React from "react";

const Message = (msg) => {
    return (
        <h1 className="msg">{msg}</h1>
    )
}

export default Message