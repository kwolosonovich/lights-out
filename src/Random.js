const Random = () => {
  const num = Math.random();
  if (num > 0.5) {
    return true;
  } else {
    return false;
  }
};

export default Random;
