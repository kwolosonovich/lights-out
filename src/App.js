import React from "react";
import Board from "./Board";
import "./App.css";
import Home from './Home'
import Footer from './Footer'

/** Simple app that just shows the LightsOut game. */

function App() {
  return (
    <div className="flex-container">
      <div className="App">
        <Board />
      </div>
      <div className="col2">
        <Home />
      </div>
      <div>
        <Footer />
      </div>
    </div>
  );
}

export default App;
