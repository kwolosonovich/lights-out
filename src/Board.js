import React, { useState } from "react";
import Cell from "./Cell";
import "./Board.css";
import Random from "./Random";
import Message from "./Message";

//  game board properties:
//  number of rows of board
const nrows = new Array(5);
//  number of cols of board
const ncols = new Array(5);
// chance any cell is lit at start of game
const chanceLightStartsOn = 0.5;



// winner message ***************************** need to fix
const msg = ""

// create a board nrows high/ncols wide, each cell randomly lit or unlit
function Board() {
  const [board, setBoard] = useState(createBoard(nrows, ncols));

  function createBoard() {
    let initialBoard = [];

    // array-of-arrays of true/false values
    for (let y = 0; y < nrows.length; y++) {
      let rowVals = [];
      for (let x = 0; x < ncols.length; x++) {
        ncols[x] = Random();

        if (ncols[x] > chanceLightStartsOn) {
          rowVals.push("on");
        } else {
          rowVals.push("off");
        }
      }

      initialBoard.push(rowVals);
    }
    return initialBoard;
  }

  // check for all cells off   **************** need to change to check for 'off'
  function hasWon() {
    return board.every((row) => row.every((cell) => !cell));
  }


  // **** Need to fix this function to account for on/off variables to flip each card****

  function flipCellsAround(cellCoords) {
    setBoard((oldBoard) => {
      const [y, x] = cellCoords.split("-").map(Number);

      console.log(cellCoords);

      const flipCell = (y, x, boardCopy) => {
        // if this coord is actually on board, flip it
        if (x >= 0 && x < ncols && y >= 0 && y < nrows) {
            boardCopy[y][x] = !boardCopy[y][x];
         }
      };

      // (deep) copy of the oldBoard
      const boardCopy = oldBoard.map((row) => [...row]);

      // flip this cell and the cells around it
      flipCell(y, x, boardCopy);
      flipCell(y, x - 1, boardCopy);
      flipCell(y, x + 1, boardCopy);
      flipCell(y - 1, x, boardCopy);
      flipCell(y + 1, x, boardCopy);

      // return the copy
      return boardCopy;
    });
  }

  // ************************ how to return message 

  // if the game is won render Message
  if (hasWon()) {
    msg = Message("You Win!")
    return msg
  }

  //create table board
  let tableBoard = [];

  for (let y = 0; y < nrows.length; y++) {
    let tableRow = [];
    for (let x = 0; x < ncols.length; x++) {
      let cellCoords = `${x}-${y}`;
      tableRow.push(
        <Cell
          key={cellCoords}
          light={board[x][y]}
          flipCellsAroundMe={() => flipCellsAround(cellCoords)}
        />
      );
    }
    tableBoard.push(
      <tr key={y} className="tableRow">
        {tableRow}
      </tr>
    );
  }

  return (
    <div className="game">
      <h1>Lights Out</h1>
      <table>
        <tbody>{tableBoard}</tbody>
      </table>
      <div>{msg}</div>
    </div>
  );
}

export default Board;
