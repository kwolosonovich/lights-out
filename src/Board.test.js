import React from 'react';
import { render } from "@testing-library/react";
import App from './App'
import Board from './Board'
import Random from './Random';
import Message from './Message';


// smoke test
it("renders without crashing", () => {
  render(<Board />);
});

// snapshot test
it("matches snapshot", () => {
    const {asFragment} = render(<Board />)
    expect (asFragment()).toMatchSnapshot()
})

