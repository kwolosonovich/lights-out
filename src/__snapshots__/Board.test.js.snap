// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="game"
  >
    <h1>
      Lights Out
    </h1>
    <table>
      <tbody>
        <tr
          class="tableRow"
        >
          <td
            alt="select"
            class="Cell off"
            role="button"
          />
          <td
            alt="select"
            class="Cell on"
            role="button"
          />
          <td
            alt="select"
            class="Cell on"
            role="button"
          />
          <td
            alt="select"
            class="Cell off"
            role="button"
          />
          <td
            alt="select"
            class="Cell on"
            role="button"
          />
        </tr>
        <tr
          class="tableRow"
        >
          <td
            alt="select"
            class="Cell off"
            role="button"
          />
          <td
            alt="select"
            class="Cell on"
            role="button"
          />
          <td
            alt="select"
            class="Cell on"
            role="button"
          />
          <td
            alt="select"
            class="Cell on"
            role="button"
          />
          <td
            alt="select"
            class="Cell on"
            role="button"
          />
        </tr>
        <tr
          class="tableRow"
        >
          <td
            alt="select"
            class="Cell off"
            role="button"
          />
          <td
            alt="select"
            class="Cell off"
            role="button"
          />
          <td
            alt="select"
            class="Cell off"
            role="button"
          />
          <td
            alt="select"
            class="Cell on"
            role="button"
          />
          <td
            alt="select"
            class="Cell on"
            role="button"
          />
        </tr>
        <tr
          class="tableRow"
        >
          <td
            alt="select"
            class="Cell off"
            role="button"
          />
          <td
            alt="select"
            class="Cell on"
            role="button"
          />
          <td
            alt="select"
            class="Cell off"
            role="button"
          />
          <td
            alt="select"
            class="Cell off"
            role="button"
          />
          <td
            alt="select"
            class="Cell on"
            role="button"
          />
        </tr>
        <tr
          class="tableRow"
        >
          <td
            alt="select"
            class="Cell on"
            role="button"
          />
          <td
            alt="select"
            class="Cell off"
            role="button"
          />
          <td
            alt="select"
            class="Cell on"
            role="button"
          />
          <td
            alt="select"
            class="Cell off"
            role="button"
          />
          <td
            alt="select"
            class="Cell on"
            role="button"
          />
        </tr>
      </tbody>
    </table>
  </div>
</DocumentFragment>
`;
